presto-hive/src/main/java/com/facebook/presto/hive/HdfsConfiguration.java

```diff
@@ -13,184 +13,11 @@
-
-        // only enable short circuit reads if domain socket path is properly configured
-        if (!config.get("dfs.domain.socket.path", "").trim().isEmpty()) {
-            config.setBooleanIfUnset("dfs.client.read.shortcircuit", true);
-        }
-
-        config.setInt("dfs.socket.timeout", Ints.checkedCast(dfsTimeout.toMillis()));
-        config.setInt("ipc.ping.interval", Ints.checkedCast(dfsTimeout.toMillis()));
-        config.setInt("ipc.client.connect.timeout", Ints.checkedCast(dfsConnectTimeout.toMillis()));
-        config.setInt("ipc.client.connect.max.retries", dfsConnectMaxRetries);
-
-        // re-map filesystem schemes to match Amazon Elastic MapReduce
-        config.set("fs.s3.impl", PrestoS3FileSystem.class.getName());
-        config.set("fs.s3n.impl", PrestoS3FileSystem.class.getName());
-        config.set("fs.s3bfs.impl", "org.apache.hadoop.fs.s3.S3FileSystem");
-
-        // set AWS credentials for S3
-        for (String scheme : ImmutableList.of("s3", "s3bfs", "s3n")) {
-            if (s3AwsAccessKey != null) {
-                config.set(format("fs.%s.awsAccessKeyId", scheme), s3AwsAccessKey);
-            }
-            if (s3AwsSecretKey != null) {
-                config.set(format("fs.%s.awsSecretAccessKey", scheme), s3AwsSecretKey);
-            }
-        }
-
-        // set config for S3
-        config.setBoolean(PrestoS3FileSystem.S3_SSL_ENABLED, s3SslEnabled);
-        config.setInt(PrestoS3FileSystem.S3_MAX_CLIENT_RETRIES, s3MaxClientRetries);
-        config.setInt(PrestoS3FileSystem.S3_MAX_ERROR_RETRIES, s3MaxErrorRetries);
-        config.set(PrestoS3FileSystem.S3_MAX_BACKOFF_TIME, s3MaxBackoffTime.toString());
-        config.set(PrestoS3FileSystem.S3_MAX_RETRY_TIME, s3MaxRetryTime.toString());
-        config.set(PrestoS3FileSystem.S3_CONNECT_TIMEOUT, s3ConnectTimeout.toString());
-        config.set(PrestoS3FileSystem.S3_SOCKET_TIMEOUT, s3SocketTimeout.toString());
-        config.set(PrestoS3FileSystem.S3_STAGING_DIRECTORY, s3StagingDirectory.toString());
-        config.setInt(PrestoS3FileSystem.S3_MAX_CONNECTIONS, s3MaxConnections);
-        config.setLong(PrestoS3FileSystem.S3_MULTIPART_MIN_FILE_SIZE, s3MultipartMinFileSize.toBytes());
-        config.setLong(PrestoS3FileSystem.S3_MULTIPART_MIN_PART_SIZE, s3MultipartMinPartSize.toBytes());
-
-        updateConfiguration(config);
-
-        return config;
-    }
-
-    @SuppressWarnings("UnusedParameters")
-    protected void updateConfiguration(Configuration config)
-    {
-        // allow subclasses to modify configuration objects
-    }
-
-    public static class NoOpDNSToSwitchMapping
-            implements DNSToSwitchMapping
-    {
-        @Override
-        public List<String> resolve(List<String> names)
-        {
-            // dfs client expects an empty list as an indication that the host->switch mapping for the given names are not known
-            return ImmutableList.of();
-        }
+    boolean verifyChecksum();
```

Comment:
It is a bit weird to have this method on this class.  The implementation is not really configuration specific and is simply a passthrough to the `HiveClientConfig`.

If you look at the only usage of this method, it is in a class that is `@Injected` and could easily take the `HiveClientConfig` directly.